### ISSUE TITLE
Building deployments from the CLI only uploads the deployment.yaml and not the whole folder 

### ISSUE THREAD

**Issue by GeorgiYGanchev** [PrefectHQ/prefect#11245]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

Hello,

I'm using Prefect Cloud and local prefect version 2.14.2.

For some reason when I build my deployments the only thing uploaded to the remote block storage is the deployment.yaml and other files from the current directory. I noticed it just because changes I've been doing to my flows had no effect and was staring to think there is something wrong with prefect itself.

It's interesting that the build itself finished normally and even uploads the deployment.yaml file every time. When I checked in the block itself (gsc) I confirmed that indeed only the YAML has been updated. The first thing that came to mind was the .prefectignore file but there is nothing there that prevents uploading .py files. You can also see that I'm not skipping the upload step in the CLI (│ --skip-upload)

Of course the workaround is easy - I manually upload the updated scripts to the block.

Any help would be appreciated. Thank!

### Reproduction

```python3
prefect deployment build ./flow.py:flow -n my-flow -q queue --storage-block gcs/gcp-block
```

### Versions

```Text
2.14.2
```

**Comment by zzstoatzz** [comment]
hello @GeorgiYGanchev - thank you for the issue!

Can you explain how this differs from your expectation (in terms of which files are uploaded)?
> when I build my deployments the only thing uploaded to the remote block storage is the deployment.yaml and other files from the current directory.

it would also be helpful if you could share your `.prefectignore` file, just in case.

**Comment by GeorgiYGanchev** [comment]
Ah, sorry I meant -  the deployment.yaml and NO other files from the current folder.

I have found the issue - I set a different working directory (using and ENV variable) in the script based on the current machine (VM) I'm using, so the deployemnt.yaml file was being exported in a different location. 
I did not expect this  to affect where the deployment file is exported but nevertheless the issue is fixed.

Thanks!
